build.go

```diff
@@ -172,50 +220,123 @@ func runTestCommandLine(pkg string, tags []string) []string {
 	if mg.Verbose() {
 		args = append(args, "-v")
 	}
-	args = append(args, formatTags(tags)...)
+	args = append(args, formatTags(goTagOpt, tags)...)
 	return append(args, pkg)
 }
 
-// RunTest runs the specified package's tests.
-func RunTest(ctx context.Context, pkg string, tags ...string) error {
-	mg.CtxDeps(ctx, BuildTestDep(pkg, tags...))
-
-	return sh.RunV(mg.GoCmd(), runTestCommandLine(pkg, tags)...)
-}
-
-type testRunner struct {
+// TestRunner implements [mg.Fn] to build (as by [BuildTest]) and run the test binary for a package using "go test."
+type TestRunner struct {
```

Comment:
Is there value in publishing this type? Seems like we only need to publish the "all" targets.